tests/rate_limited.rs

```diff
@@ -0,0 +1,325 @@
+use proxy_wasm_test_framework::tester;
+use proxy_wasm_test_framework::types::{Action, BufferType, LogLevel, MapType, ReturnType};
+use serial_test::serial;
+use std::path::Path;
+
+fn wasm_module() -> String {
+    let wasm_file = Path::new("target/wasm32-unknown-unknown/release/wasm_shim.wasm");
```

Comment:
This means that the code tested is the actual (byte)code compiled with `wasm32-unknown-unknown` build target? 

Weird, but good enough for now I guess.

Reply:
Yes it is… Sorry if that was unclear, it's something I tried to explain early on when I first share this:

This embeds a runtime, [wasmtime](https://crates.io/crates/wasmtime), loads the wasm module and "pretends" to be a host, thru the [test-framework](https://github.com/Kuadrant/wasm-test-framework). It is complete blackbox testing, so we can't put the underlying code under test… we _could_ do assertion about the state of the memory in the module (the same was as the test framework reads & write to the wasm module's memory), but I think that'd be pushing it too far.